src/flow.jl

```diff
@@ -283,3 +415,423 @@ end
+                    ηdst,
+                    sw.hf[i],
+                    sw.a[i],
+                    sw.r[i],
+                    sw.dl_at_link[i],
+                    sw.mannings_n[i],
+                    sw.g,
+                    sw.froude_limit,
+                    Δt,
+                )
+            else
+                sw.q[i] = 0.0
+            end
+            sw.q_av[i] += sw.q[i] * Δt
+        end
+    end
+    if update_h
+        @threads for i = 1:sw.n
+            if sw.reservoir_index[i] == 0 || sw.lake_index[i] == 0
+                sw.volume[i] =
+                    sw.volume[i] +
+                    (
+                        sum_at(sw.q, links_at_node.src[i]) -
+                        sum_at(sw.q, links_at_node.dst[i]) + sw.inwater[i]
+                    ) * Δt
+                if sw.volume[i] < 0.0
+                    sw.error[i] = sw.error[i] + abs(sw.volume[i])
+                    sw.volume[i] = 0.0 # set volume to zero
+                end
+                sw.volume[i] = max(sw.volume[i] + sw.inflow[i] * Δt, 0.0) # add external inflow
+                sw.h[i] = sw.volume[i] / (sw.dl[i] * sw.width[i])
+                sw.h_av[i] += sw.h[i] * Δt
+            end
+        end
+    end
+end
+
+function update(
+    sw::ShallowWaterRiver,
+    network;
+    inflow_wb = nothing,
+    doy = 0,
+    update_h = true,
+)
+    @unpack nodes_at_link, links_at_node = network
+
+    if !isnothing(sw.reservoir)
+        sw.reservoir.inflow .= 0.0
+        sw.reservoir.totaloutflow .= 0.0
+    end
+    if !isnothing(sw.lake)
+        sw.lake.inflow .= 0.0
+        sw.lake.totaloutflow .= 0.0
+    end
+    sw.q_av .= 0.0
+    sw.h_av .= 0.0
+
+    t = 0.0
+    while t < sw.Δt
+        Δt = stable_timestep(sw)
+        if t + Δt > sw.Δt
+            Δt = sw.Δt - t
+        end
+        shallowwater_river_update(sw, network, Δt, inflow_wb, doy, update_h)
+        t = t + Δt
+    end
+    sw.q_av ./= sw.Δt
+    sw.h_av ./= sw.Δt
+end
+
+
+"""
+    stable_timestep(sw::ShallowWaterRiver)
+
+Compute a stable timestep size for the local inertial approach for river flow, based on
+Bates et al. (2010).
+
+Δt = α * (Δx / sqrt(g max(h))
+"""
+
+function stable_timestep(sw::ShallowWaterRiver)
+    Δtₘᵢₙ = Inf
+    for i = 1:sw.n
+        Δt = sw.α * sw.dl[i] / sqrt(sw.g * sw.h[i])
+        Δtₘᵢₙ = Δt < Δtₘᵢₙ ? Δt : Δtₘᵢₙ
+    end
+    Δtₘᵢₙ = isinf(Δtₘᵢₙ) ? 10.0 : Δtₘᵢₙ
+    return Δtₘᵢₙ
+end
+
+"""
+    set_river_inwater(model::Model{N,L,V,R,W,T}, ssf_toriver) where {N,L,V<:SBM,R,W,T}
+
+Set `inwater` of the river component for a `Model` with vertical `SBM` concept.
+`ssf_toriver` is the subsurface flow to the river.
+"""
+function set_river_inwater(model::Model{N,L,V,R,W,T}, ssf_toriver) where {N,L,V<:SBM,R,W,T}
+    @unpack lateral, vertical, network = model
+    inds = network.index_river
+
+    @. lateral.river.inwater = (
+        ssf_toriver .+ lateral.land.to_river[inds] +
+        # net_runoff_river
```

Comment:
Can be removed?

Reply:
Not sure what you mean here, why could this part be removed?